pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -298,13 +300,16 @@ public synchronized void readEntriesComplete(List<Entry> entries, Object ctx) {
             int messagesForC = Math.min(Math.min(entriesToDispatch, c.getAvailablePermits()), MaxRoundRobinBatchSize);
 
             if (messagesForC > 0) {
-                int msgSent = c.sendMessages(entries.subList(start, start + messagesForC)).getRight();
-
+                
+                // remove positions first from replay list first : sendMessages recycles entries
```

Comment:
it is related to this PR. in this change we remove entries from replay list before we send those entries to consumer. Because it's consumer's responsibility to release entry's data and recycle it. 
Now, to remove positions from replay list after sending to consumer, we have to read positions from already recycled entires which gives NPE.
So, we remove first from replay list and then send to consumer.